lib/flavor-downloader.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Created by AsherS on 8/20/15.
+ */
+
+var Q = require('q');
+var _ = require('underscore');
+var path = require('path');
+var url = require('url');
+var httpUtils = require('./utils/HTTPUtils');
+var qfs = require("q-io/fs");
+var logger = require('./logger/logger');
+var config = require('./Configuration');
+var m3u8Handler = require('./promise-m3u8');
+var chunklistM3u8Generator = require('./ChunklistManifestGenerator');
+
+var FlavorDownloader = (function () {
+
+    function HLSDownloader(m3u8Url, destPath) {
+        this.streamUrl = m3u8Url;
+        this.destFolderPath = destPath;
+        this.shouldRun = true;
+        this.pollingInterval = config.get("pollingInterval");
+        this.tmpM3u8Folder = path.join(destPath, 'tmpM3u8');
+        this.playlistName = path.basename(url.parse(m3u8Url).pathname);
+        this.m3u8generator = chunklistM3u8Generator(this.destFolderPath, this.playlistName, config.get("dvrWindow"));
```

Comment:
playlistName is calculated based on the *live* m3u8Url. How will it get the correct manifest name originating from PersistenceFormat? What am I missing?